visualime/visualize.py

```diff
@@ -318,6 +318,66 @@ def scale_opacity(
     return new_overlay
 
 
+def mark_boundaries(
+        image: np.ndarray,
+        segment_mask: np.ndarray,
+        color: Union[str, Tuple[int, int, int]] = "red",
+        opacity: float = 1.0,
+) -> np.ndarray:
+    """Mark the boundaries of the segments in the image.
+
+    Parameters
+    ----------
+    image : np.ndarray
+        The image to mark the boundaries of the segments in.
+        Can be either a grayscale or an RGB image.
+        The image will be modified in-place.
+
+    segment_mask : np.ndarray
+        The mask generated by :meth:`visualime.lime.create_segments`:
+        An array of shape `(image_width, image_height)`.
+
+    color : str or int 3-tuple (RGB), default "red"
+        The color for the boundaries.
+        Can be given as a color name or an RGB tuple.
+
+        Color names are parsed through :meth:`PIL.ImageColor.getrgb`.
+        To obtain a list of available color names, run:
+
+        >>> from PIL.ImageColor import colormap, getrgb
+        >>> for name, code in colormap.items(): print(name, getrgb(code))
+
+        Note that while it is possible to pass an RGBA tuple, only the RGB values
+        will be considered, the opacity will be determined by the `opacity` parameter.
+
+    opacity : float, default 1.0
+        The opacity of the boundaries as a number between `0.0` and `1.0`.
+
+    Returns
+    -------
+    np.ndarray
+        An array of shape `(image_width, image_height, 3)` representing an RGB image.
+    """
+    rgb_color = _get_color(color, opacity)[:3]
+
+    # check if the shape of the mask fits the image
+    if image.shape[:2] != segment_mask.shape:
+        raise ValueError("The shape of the mask does not match the shape of the image.")
+
+    # check if the image is grayscale or RGB
```

Comment:
```suggestion
```

Das ist ein unnötiger Kommentar, das ergibt sich direkt aus der nächsten Zeile.

Die Frage ist allerdings, warum ausgerechnet hier in dieser Funktion ein Graustufen-Bild in ein RGB-Bild konvertiert werden soll. Das machen wir sonst nirgendwo, oder? VisuaLIME unterstützt durchgehend RGB-Bilder. Es gibt eine ganze Reihe von Funktionen, die mit Graustufen-Bildern nicht funktionieren, glaube ich.

Mein Vorschlag ist, die Graustufen-Bilder erst einmal weg zu lassen und dann ggf. in einem zweiten Schritt Support dafür zu ergänzen, dann aber gleich überall.